### DISCUSSION TITLE
resource data

### DISCUSSION THREAD

**Discussion by beze20** [alvarobartt/investpy#265] [Q&A] [answered]
Hi alvarobartt!

Thank you for investpy I was happy when I found it.
 
I would like to ask you how did you generate/save the .csv files with the stocks/funds etc data? 

As I see investpy use the tickers from the saved .csv files, I'm right? As I compared the number of US stocks on investing.com I found x3 more stocks. I check some stocks and my guess is that the stocks from the OTC market aren't saved in the .csv. And I also found some stocks which are missing from the .csv. 

I tried to scrap the tickers from the website but I wasn't able to do that (with BeautifulSoup). Did you tried it?

I would like to use investpy with all stocks from the US and EU markets, is it possible?

Thank you!

Best regards,
Zoltan

**Comment by alvarobartt** [comment, marked as answer]
Hi @beze20! 😄 

I'm glad you liked investpy and found it useful. Currently the script for retrieving the data is private, but it basically sends requests to Investing.com internal API so as to retrieve all the meta information from the website. You can manually add them in the CSV files including all the fields or you can open an issue describing which ones are missing with the website URL for each of them attached.

Thank you also for using this Dicussions page! Feel free to follow me both on GitHub and on Twitter at @alvarobartt 👍🏻

**Comment by MartinW90** [comment]
Hi Alvarobartt!

First of all I really appreciate your project. I've learned a lot reading your code and it is really useful for a project I'm currently working on. 
I need historic stock data information from a symbol not in your list. I found that only the curr_id war required to retrieve the data. This id is also returned by using your search function. So instead of the lookup in the file, I called your search function. This enables me to get data for every symbol available on investing.com. (in a first smoke test:p) So my question is: Do I miss something? Why did you decide to go for the csv approach?

Again thanks a lot for your work!
regards,
Martin

**Reply by alvarobartt** [reply]
Hi @MartinW90 thanks for your kind words!

As you say the ID is the unique identifier of every symbol and, so on, the one used in the request parameters. And also, the investpy.search_quotes() function contains all the Investing.com financial products, but it has some latency attached. That is mainly why I decided to create a CSV files containing the financial products indexed on Investing.com, but due to some Investing.com internal issues (which I cannot solve) not all the financial products are that easily accessible while combining the search filters, but whenever you use their search engine all the financial products are present there.

So to sum up, I could use the search engine to search any symbol (or the top match) instead of using the CSV, but that will increase the latency! 😩 Anyway I think that the search function is clear enough and in case the financial product is not available on the CSV files you can easily look for it using the search engine and then retrieve the historical data from the resulting SearchObj.

P.S.: I think next time you can open a new discussion because it's easier for me to manage! 🤗

**Reply by MartinW90** [reply]
Hey Alvarobartt,
thanks a lot for your fast reply:) Doesn't it also take ages to search through the CSV file? An approach might be to put the .search_quotes() function into the the catch block of the CSV search and then enhance the file, so the next time you go for a certain product the search isn't necessary anymore.
Sorry, I'm new :p I'll open a new discussion the next time

Regards,
Martin
